rootfs/etc/haproxy/template/haproxy.tmpl

```diff
@@ -160,6 +169,13 @@ frontend httpsfront
 {{ else }}
     use_backend httpsback-{{ labelize $server.Hostname }} if { req.ssl_sni -i {{ $server.Hostname }} }
 {{ end }}
+{{ if ne $server.Alias "" }}
+{{ if isRegexpHostname $server.Alias }}
+    use_backend httpsback-{{ labelize $server.Hostname }} if { req.ssl_sni -m reg -i {{ aliasRegex $server.Alias }} }
+{{ else }}
+    use_backend httpsback-{{ labelize $server.Hostname }} if { req.ssl_sni -i {{ $server.Alias }} }
+{{ end }}
+{{ end }}
```

Comment:
About `aliasRegex`, what about remove it altogether? This would avoid issues like trying to use the `.` as a regex instead of as a literal. If so you could also simplify the expression above and use `isRegexpHostname` inline - check in a single line if `-m reg` will/won't be used.

Reply:
Actually I was too sleepy and I thought for some reason that it is a good idea to have this function.

We have two ways of using this annotation:
1. Hostname as a regex.
2. Hostname as a plain domain.

Both ways could be checked by `isRegexpHostname`.

So if we single-quote all the regexes in the template, then we can get rid of `aliasRegex` from all the places where it is used.

What do you think?

Reply:
No, we can't. Annotation doesn't accept backslashes so I had to keep this function.